client/v2/client.go

```diff
@@ -248,11 +269,55 @@ func (p *Point) Fields() map[string]interface{} {
 	return p.pt.Fields()
 }
 
+func (uc *udpclient) Write(bp BatchPoints) error {
+	var addr *net.UDPAddr
+	var con *net.UDPConn
+	addr, err := net.ResolveUDPAddr("udp", uc.addr)
+	if err != nil {
+		return err
+	}
+
+	con, err = net.DialUDP("udp", nil, addr)
+	if err != nil {
+		return err
+	}
+	defer con.Close()
+	if err = con.SetWriteBuffer(UDPBufferSize); err != nil {
+		return err
+	}
+
+	var b bytes.Buffer
+	var d time.Duration
+	d, _ = time.ParseDuration("1" + bp.Precision())
+
+	for _, p := range bp.Points() {
+		if _, err := b.WriteString(p.pt.RoundedString(d)); err != nil {
+			return err
+		}
+
+		if err := b.WriteByte('\n'); err != nil {
+			return err
+		}
+
+		// Write and reset the buffer if we reach the max size
+		if b.Len() > UDPMaxSendSize {
```

Comment:
While this will probably work in practise, this is not watertight. Instead the logic should be checking that *if* adding the new point to the existing buffer will send the buffer over the limit, then call `con.Write()` before adding the new point.

Make sense?